File: nats-test/python/src/events/ticket_created_listener.py
from .subjects import Subjects
from .base_listener import Listener
from .ticket_created_event import TicketCreatedEventData
import asyncio


class TicketCreatedListener(Listener[TicketCreatedEventData]):
    subject = Subjects.TicketCreated
    queue_group_name = 'ticket-service-python'

    async def on_message(self, data: TicketCreatedEventData, msg, sc):
        print(data)
        await sc.ack(msg)
        # return await super().on_message

File: nats-test/js/src/events/ticket-created-listener.ts
import { Message } from "node-nats-streaming";
import { Listener } from "./base-listener";
import { Subjects } from "./subjects";
import { TicketCreatedEvent } from "./ticket-created-event";

export class TicketCreatedListener extends Listener<TicketCreatedEvent> {
  subject: Subjects.TicketCreated = Subjects.TicketCreated;

  queueGroupName = "payment-service-javascript";
  onMessage(data: TicketCreatedEvent["data"], msg: Message) {
    console.log("Event received: ", data);

    console.log(data.id);
    console.log(data.title);
    console.log(data.price);

    msg.ack();
  }
}
